File: backend/app/models/workout_session.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WorkoutSession(BaseModel):
    id: str
    exercise_id: str
    exercise_name: str
    performed_at: datetime
    sets_completed: int
    reps_completed: str
    notes: str
    created_at: datetime


class WorkoutSessionCreate(BaseModel):
    exercise_id: str = Field(min_length=1)
    performed_at: datetime
    sets_completed: int = Field(ge=1, le=20)
    reps_completed: str = Field(min_length=1, max_length=50)
    notes: str = Field(default="", max_length=500)


class WorkoutSessionPayload(BaseModel):
    ok: bool = True
    data: WorkoutSession


class WorkoutSessionListPayload(BaseModel):
    ok: bool = True
    data: list[WorkoutSession]

File: flutter_app/lib/features/workout_sessions/domain/workout_session.dart
class WorkoutSession {
  final String id;
  final String exerciseId;
  final String exerciseName;
  final DateTime performedAt;
  final int setsCompleted;
  final String repsCompleted;
  final String notes;

  const WorkoutSession({
    required this.id,
    required this.exerciseId,
    required this.exerciseName,
    required this.performedAt,
    required this.setsCompleted,
    required this.repsCompleted,
    required this.notes,
  });

  factory WorkoutSession.fromJson(Map<String, dynamic> json) {
    return WorkoutSession(
      id: json['id'] as String? ?? '',
      exerciseId: json['exercise_id'] as String? ?? '',
      exerciseName: json['exercise_name'] as String? ?? 'Ejercicio',
      performedAt: DateTime.tryParse(json['performed_at'] as String? ?? '') ??
          DateTime.now(),
      setsCompleted: json['sets_completed'] as int? ?? 0,
      repsCompleted: json['reps_completed'] as String? ?? '-',
      notes: json['notes'] as String? ?? '',
    );
  }
}
